src/fparser/two/tests/fortran2003/test_program_r201.py

```diff
@@ -0,0 +1,326 @@
+def test_missing_prog_multi(f2003_create):
+    '''Test that a main program program_unit without a program declaration
+    can be parsed successfully when it is not the first program_unit.
+
+    '''
+    reader = get_reader('''\
+      subroutine first
+      end
+      end
+      ''')
+    ast = Program(reader)
+    assert "SUBROUTINE first\n" \
+        "END SUBROUTINE" in str(ast)
+    assert "END PROGRAM" in str(ast)
+
+# A program should contain only only one main program
+
+
+@pytest.mark.xfail(reason="Only one main program is allowed in a program")
+def test_one_main1(f2003_create):
+    '''Test that multiple main programs raise an exception.'''
+    reader = get_reader('''\
+      program first
+      end
+      program second
+      end
+      ''')
+    with pytest.raises(FortranSyntaxError) as excinfo:
+        dummy_ = Program(reader)
+    assert ("XXX"
+            in str(excinfo.value))
+
+
+# Check comments are supported at this level
+
+
+def test_comment0(f2003_create):
+    '''Test that a single program_unit without comments can be parsed
+    successfully with comment processing switched on.
+
+    '''
+    reader = get_reader((
+        "subroutine test()\n"
+        "end subroutine\n"), ignore_comments=False)
+    ast = Program(reader)
+    assert ("SUBROUTINE test\n"
+            "END SUBROUTINE test") in str(ast)
+
+
+def test_comment1(f2003_create):
+    '''Test that a single program_unit can be parsed successfully with
+    comments being ignored.'''
+    reader = get_reader('''\
+      ! comment1
+      subroutine test()
+      end subroutine
+      ! comment2
+      ''')
+    ast = Program(reader)
+    assert "SUBROUTINE test\n" \
+        "END SUBROUTINE" in str(ast)
+    assert "! comment" not in str(ast)
+
+
+def test_comment2(f2003_create):
+    '''Test that a single program_unit can be parsed successfully with
+    comments being included.'''
+    reader = get_reader((
+        "! comment1\n"
+        "subroutine test()\n"
+        "end subroutine\n"
+        "! comment2\n"), ignore_comments=False)
+    ast = Program(reader)
+    assert ("! comment1\n"
+            "SUBROUTINE test\n"
+            "END SUBROUTINE test\n"
+            "! comment2") in str(ast)
+
+
+def test_comment3(f2003_create):
+    '''Test that multiple program_units can be parsed successfully with
+    comments being ignored.'''
+    reader = get_reader('''\
+      ! comment1
+      subroutine test()
+      end subroutine
+      ! comment2
+      module example
+      end module
+      ! comment3
+      ''', ignore_comments=True)
+    ast = Program(reader)
+    assert ("SUBROUTINE test\n"
+            "END SUBROUTINE test\n"
+            "MODULE example\n"
+            "END MODULE example") in str(ast)
+    assert "! comment" not in str(ast)
+
+
+def test_comment4(f2003_create):
+    '''Test that multiple program_units can be parsed successfully with
+    comments being included.'''
+    reader = get_reader((
+      "! comment1\n"
```

Comment:
pylint complains about the indentation here and at line 318.

Reply:
Fixed.